keyboards/unicomp/mini_m/minirazz/info.json

```diff
@@ -0,0 +1,29 @@
+{
+    "manufacturer": "Unicomp",
+    "maintainer": "purdeaandrei",
+    "url": "https://github.com/purdeaandrei/MiniRazz",
+    "usb": {
+        "device_version": "0.0.1",
+        "pid": "0x27DB",
+        "vid": "0x16C0"
```

Comment:
This vendor ID is already in use.
https://www.the-sz.com/products/usbid/index.php?v=0x16C0

Reply:
This vendor ID is provided for free as shared usb id by v-usb:
https://github.com/obdev/v-usb/blob/master/usbdrv/USB-IDs-for-free.txt

Reply:
This is still an issue as it is not a unique VID/PID combo, and if you or community users wish to use VIA or similar could be considered a blocker.

Reply:
I am not planning on supporting VIA ever.
I am supporting VIAL however, but that doesn't require a unique VID/PID. VIAL detects the keyboard by a magic number embedded somewhere in the SERIAL_NUMBER. (Which for VIAL it means that I have to have a serial number like "purdea.ro:minirazz:MAGIC_NUMBER")

Reply:
Sorry, i don't mean to annoying by refusing to change something, just let me explain my thinking of why I'm resisting this:

If anyone else from my users isn't satisfied with VIAL, and wants to add 
VIA support they can of course get a different, unique pair of VID:PID in the future, I'm not gonna block that.

I don't see why I should bother to go requesting a PID code from https://pid.codes/ now, since I don't need it, and occupying a valuable PID code for no reason is wasteful. I am not a keyboard manufacturer, just an open source contributor, I can't afford buying a USB VID, so I must rely on one of the free offerings available.

XAP Also doesn't need Unique VID:PID so hopefully need for unique vid:pid will disappear even for VIA users in the future, so hopefully my hypothetical future users who will want to add VIA support will not need to acquire a vid:pid pair after all.

Reply:
I'm not sure that using the 16C0:27DB USB ID for a QMK-based device is proper, because QMK can present more than just a HID keyboard device to the host:
* HID mouse or joystick — these have separate entries in the ID table;
* HID digitizer (absolute pointer) — this one would probably be considered a “generic HID class device”;
* HID interfaces used by console, VIA, Vial or XAP — probably “generic HID class device” too;
* MIDI — has a separate entry in the ID table, and uses a completely different driver on Windows;
* CDC-ACM serial — also has a separate entry in the ID table, and uses a completely different driver on Windows.

These non-HID options are probably the most problematic, because apparently Windows caches the driver configuration by `idVendor:idProduct:bcdDevice`, and can end up using a completely wrong driver if the cached configuration does not match the actual interfaces presented by the device.

Also apparently there are already some users of 16C0:27DB in the repo:
- `dm9records/plaid`
- `dm9records/tartan`
- `ibm/model_m_4th_gen`
- `ibm/model_m_4th_gen/overnumpad_1xb`
- `keyhive/lattice60`
- `ortho5by12`
- `silverbullet44`
- `unicomp/classic_ultracl_post_2013`
- `unicomp/classic_ultracl_pre_2013`
- `unicomp/pc122`
- `unicomp/spacesaver_m_post_2013`
- `unicomp/spacesaver_m_pre_2013`

Reply:
The `unicomp/*` and `ibm/*` are mine. I can make changes to it if it becomes necessary. The others are not.

`dm9records/tartan` with `16c0:27db` actually has via support: https://github.com/the-via/keyboards/pull/526
This isn't really correct, since it's a shared vid:pid, and I don't think there's a way for VIA to differentiate. `dm9records/plaid` has a via keymap, but no matching VIA json in the `the-via/keyboards` repository.

When it comes to compliance with the V-USB terms, I think using 16c0/27db is okay: Under the section "IDs FOR DISCRIMINATION BY SERIAL NUMBER", it says "The USB device MUST provide a textual representation of the serial number, unless ONLY the operating system's default class driver is used.", this implies that both vendor class and an OS default class can coexist, as long as the serial number is specified. It is indeed true that combinations of different kinds of HID is not explicitly permitted, but based on the above I don't think they intended to disallow it. They definitely don't use the precise language of 'usage'/'usage page' that HID documentation uses, instead the say 'Description of use'. I read this as a much fuzzier specification, and even though the keyboard can be compiled to act as a USB mouse too, it still fundamentally is a keyboard, so I think the right VID:PID for it is 16c0/27db, and I think using it this way does comply with the V-USB terms. I fired off a message to Objective Development to check if my interpretation is correct, I will report back if/when I get a response.

Regarding windows caching behavior, I am not really able to test this since I don't have any windows systems. I maybe be able to do some directed testing by borrowing a laptop from someone else. Do you have a reference to where I can read more about this behavior, to figure out how I can test for it? It seems to me like this behavior  as it has been described here so far, would affect keymaps too, enabling/disabling certain features, and would affect temporary builds too that enable the CONSOLE for debugging, so I'm surprised that I haven't heard about this. Do windows users really have to overwrite DEVICE_VER in custom keymaps, and when changing what features are enabled?

Reply:
@sigprof  I have tested the following on a borrowed windows 10 machine:

1) 16c0:27db with all features enabled (8 interfaces in use), shows up nicely, this is the view in Device Manager, if you select View > Devices By Connection:
![image](https://user-images.githubusercontent.com/544648/234136994-6cb85ca7-cd9a-48c6-ad2e-17696540f410.png)
Ableton detects the midi interface nicely.
This vid:pid has never been plugged into this machine before, so should be starting with a fresh "cache".

2) 16c0:27db disabled all of the features, it doesn't show up as a composite device anymore. Ableton doesn't detect the MIDI interface anymore
![image](https://user-images.githubusercontent.com/544648/234137078-a098cf40-a29a-4404-8fa2-1bf0105544ec.png)

3) c0ff:fffe disabled all of the features (should be starting fresh with no cache)
Same as point 2 above

4) c0ff:fffe with all features enabled
Same as point 1 above.
Ableton correctly detects the midi interface

I don't see any evidence of caching causing any kind of trouble.
If you think caching really is a problem, then could you give me some set of steps that would reproduce it, or at least some documentation that I can reference to come up with a possible theory as to how it could be causing trouble?